src/OrleansProviders/Streams/Generator/GeneratorAdapterFactory.cs

```diff
@@ -77,36 +83,33 @@ public class GeneratorAdapterFactory : IQueueAdapterFactory, IQueueAdapter, IQue
         /// </summary>
         protected Func<ReceiverMonitorDimensions, ITelemetryProducer, IQueueAdapterReceiverMonitor> ReceiverMonitorFactory;
 
+        public GeneratorAdapterFactory(string providerName, GeneratedStreamOptions options, IServiceProvider serviceProvider, SerializationManager serializationManager, ITelemetryProducer telemetryProducer, ILoggerFactory loggerFactory)
+        {
+            this.Name = providerName;
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
+            this.serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+            this.serializationManager = serializationManager ?? throw new ArgumentNullException(nameof(serializationManager));
+            this.telemetryProducer = telemetryProducer ?? throw new ArgumentNullException(nameof(telemetryProducer));
+            this.loggerFactory = loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory));
+            this.logger = loggerFactory.CreateLogger<GeneratorAdapterFactory>();
+        }
+
         /// <summary>
         /// Initialize the factory
         /// </summary>
-        /// <param name="providerConfig"></param>
-        /// <param name="providerName"></param>
-        /// <param name="svcProvider"></param>
-        public void Init(IProviderConfiguration providerConfig, string providerName,  IServiceProvider svcProvider)
+        public void Init()
         {
-            this.loggerFactory = svcProvider.GetRequiredService<ILoggerFactory>();
-            serviceProvider = svcProvider;
-            this.providerName = providerName;
-            receivers = new ConcurrentDictionary<QueueId, Receiver>();
-            adapterConfig = new GeneratorAdapterConfig(providerName);
-            adapterConfig.PopulateFromProviderConfig(providerConfig);
-            this.serializationManager = svcProvider.GetRequiredService<SerializationManager>();
-            this.telemetryProducer = svcProvider.GetService<ITelemetryProducer>();
+            this.receivers = new ConcurrentDictionary<QueueId, Receiver>();
             if (CacheMonitorFactory == null)
                 this.CacheMonitorFactory = (dimensions, telemetryProducer) => new DefaultCacheMonitor(dimensions, telemetryProducer);
             if (this.BlockPoolMonitorFactory == null)
                 this.BlockPoolMonitorFactory = (dimensions, telemetryProducer) => new DefaultBlockPoolMonitor(dimensions, telemetryProducer);
             if (this.ReceiverMonitorFactory == null)
                 this.ReceiverMonitorFactory = (dimensions, telemetryProducer) => new DefaultQueueAdapterReceiverMonitor(dimensions, telemetryProducer);
-            if (adapterConfig.GeneratorConfigType != null)
+            generatorConfig = this.serviceProvider.GetServiceByName<IStreamGeneratorConfig>(this.Name);
```

Comment:
constructor injection instead of this.

Reply:
generator config is not required, it can be provided at runtime, hence constructor injection will not work :/